### PR TITLE
Update StopOnRevertHandler.t.sol

### PR DESCRIPTION
**Pull request by touqeerShah**
redeemCollateral have problem in getCollateralBalanceOfUser function argument so it was never call that why it never give issue it define as user address then contract collateral address but pass swap of it so that why it never run and revert from if conditions

### CODE REVIEW

**Comment by PatrickAlphaC** [comment]
Ahhhh I See what you're doing here! Thank you!

So this is good, but we are just depositing and then withdrawing right away. I think a better play would be to only redeem collateral from users who have deposited. What do you think?

**Comment by touqeerShah** [comment]
Let suppose it call function randomly if it call `redeemCollateral` first without depositing it give following error that why I add deposit in the test 


```[FAIL. Reason: Arithmetic over/underflow]
        [Sequence]
                sender=0x03c82d46bad32f14f988b84e10b80577d433a676 addr=[test/fuzz/failOnRevert/StopOnRevertHandler.t.sol:StopOnRevertHandler]0x2e234dae75c793f67a35089c9d99245e1c58470b calldata=mintAndDepositCollateral(uint256,uint256), args=[47409767332275032229872235448 [4.74e28], 23614243684831295026004535944408119379691892249249472724442933517775913615360 [2.361e76]]
```

**Comment by PatrickAlphaC** [comment]
Right, but we could improve this even more by keeping track of all the people who have deposited collateral, and then call redeem collateral with only people who have deposited.

File: test/fuzz/failOnRevert/StopOnRevertHandler.t.sol [hunk 1]
```diff
@@ -57,15 +57,35 @@ contract StopOnRevertHandler is Test {
 
     function redeemCollateral(uint256 collateralSeed, uint256 amountCollateral) public {
         ERC20Mock collateral = _getCollateralFromSeed(collateralSeed);
-        uint256 maxCollateral = dscEngine.getCollateralBalanceOfUser(address(collateral), msg.sender);
+        amountCollateral = bound(amountCollateral, 1, MAX_DEPOSIT_SIZE);
+        vm.startPrank(msg.sender);
+        collateral.mint(msg.sender, amountCollateral);
+        collateral.approve(address(dscEngine), amountCollateral);
+        dscEngine.depostCollateral(address(collateral), amountCollateral);
```
Comment (PatrickAlphaC) [inline test/fuzz/failOnRevert/StopOnRevertHandler.t.sol:64]:
```suggestion
        dscEngine.depositCollateral(address(collateral), amountCollateral);
```

**Comment by PatrickAlphaC** [comment]
In any case, I like your addition

**Comment by PatrickAlphaC** [comment]
Going to close as we are now pranking msg.sender to fix the issues here

**Pull request closed without merging**